azure-app-service/massdriver.yaml

```diff
@@ -0,0 +1,195 @@
+schema: draft-07
+name: <md .Name md>
+description: <md .Description md>
+source_url: github.com/YOUR_ORG/<md .Name md>
+access: private
+type: application
+
+app:
+  envs:
+    # Use jq expressions to build environment variables from input params or connections
+    # LOG_LEVEL: .params.log_level
+    # MONGO_DSN: .connections.mongo_authentication.data.authentication.username + ":" + .connections.mongo_authentication.data.authentication.password + "@" + .connections.mongo_authentication.data.authentication.hostname + ":" + (.connections.mongo_authentication.data.authentication.port|tostring)
+  policies: []
+    # Use jq expressions to select policies from artifact security blocks
+    # - .connections.sqs.data.security.iam.subscribe
+
+params:
+  examples:
+    - __name: Development
+      application:
+        sku_name: S1
+        minimum_worker_count: 1
+        maximum_worker_count: 3
+    - __name: Production
+      application:
+        sku_name: P3v3
+        minimum_worker_count: 1
+        maximum_worker_count: 10
+  required:
+    - application
+    - repo
+    - dns
+  properties:
+    application: 
+      title: Application
+      description: Application configuration
+      type: object
+      required:
+        - location
+        - sku_name
+        - minimum_worker_count
+        - maximum_worker_count
+      properties:
+        location:
+          title: Location
+          description: Azure location
+          type: string
+          $md.immutable: true
+          oneOf:
+            - title: West US
+              const: westus
+            - title: West US 2
+              const: westus2
+            - title: West US 3
+              const: westus3
+            - title: East US
+              const: eastus
+            - title: East US 2
+              const: eastus2
+            - title: North Central US
+              const: northcentralus
+            - title: South Central US
+              const: southcentralus
+            - title: Central US
+              const: centralus
+        sku_name:
+          title: Size
+          description: Azure App Service Plan size
+          type: string
+          oneOf:
+            - title: 1 vCPU, 1.75GB RAM
+              const: S1
+            - title: 2 vCPU 8GB RAM
+              const: P1v3
+            - title: 4 vCPU 16GB RAM
+              const: P2v3
+            - title: 8 vCPU 32GB RAM
+              const: P3v3
+        minimum_worker_count:
+          title: Minimum instances
+          description: Minimum number of instances for autoscale.
+          type: integer
+          minimum: 1
+          maximum: 30
+        maximum_worker_count:
+          title: Maximum instances
+          description: Maximum number of instances for autoscale.
+          type: integer
+          minimum: 1
+          maximum: 30
+        notification_email:
```

Comment:
FYI, we actually send down the email of the contact that owns the target(?) as part of md_metadata now. `var.md_metadata.target.contact_email`.  We can ask them here, or we can just use that e-mail to simplify things.

Reply:
Would it be the same variable structure with this application template? For example, in the application module, I was encouraged to rename `var.md_metadata.name_prefix` to just `var.name`

Reply:
Removed from here and updated on the application module. Though I'm not sure how it gets passed through in the module itself.